### DISCUSSION TITLE
Clustering coefficient resulting in a complex number

### DISCUSSION THREAD

**Discussion by albitcaban** [networkx/networkx#4971] [Q&A]
Hi everyone,

When I calculate the clustering coefficient of individual nodes in an undirected weighted graph with both positive and negative edges the result ends up being a complex number (i.e. 023+0.008j). Has anyone had this issue or know how to solve it?

I'm using nx.clustering(G, weight='weight').

**Comment by dschult** [comment]
Which version of NetworkX are you using?
The negative values come from a cube root in the formula. The trouble with cube roots is that there are 3 of them... typically 2 of the 3 are complex values. Python's power function returns complex values for some cube roots when a real value is also available. The latest version of NetworkX fixes (#4519) this strange result by using `numpy.cbrt` which returns a real value for all real input values.  NetworkX 2.6 has the fix. v2.5 does not.  And NetworkX 2.6 should be ready to go soon.
